### DISCUSSION TITLE
PermissionError in ozone_redistribution

### DISCUSSION THREAD

**Discussion by mo-abodas** [MetOffice/simulation-systems#113] [UM]
Hello, I've restarted a GC suite that had run for nearly 100 years, and the ozone_redistribution app is failing with the following error:
PermissionError: [Errno 13] Permission denied: '/home/d00/hadac/cylc-run/u-cz993/share/data/ozone_redistribution/mmro3_monthly_CMIP6_2082_N96_cz993-ancil_2anc'

Has anyone encountered this issue before? Thanks!

**Comment by stevemullerworth** [comment]
The file seems to be there and is readable! 

Has it failed more than once? If it had failed once I might suggest a weird and hopefully temporary file system problem.

```

$ ls -l /projects/ancils/cmip6/ancils/n96e/timeslice_1850/OzoneConc/v1/mmro3_monthly_CMIP6_1850_N96_edited-ancil_2anc
-r-xrw-r-- 1 jwalton ancils 234504192 Mar 30  2017 /projects/ancils/cmip6/ancils/n96e/timeslice_1850/OzoneConc/v1/mmro3_monthly_CMIP6_1850_N96_edited-ancil_2anc
```

**Reply by mo-abodas** [reply]
Thanks @stevemullerworth 
Yes, it's failed more than once. Looking at the .err file I've noticed that it contains this warning about ssh:

Warning: the ECDSA host key for 'exvcylc11.metoffice.gov.uk' differs from the key for the IP address '...'
Offending key for IP in /home/h06/hadac/.ssh/known_hosts:139
Matching host key in /home/h06/hadac/.ssh/known_hosts:145

This warning doesn't appear in other suites similar to this one that I'm running. Could this be related? If so, how can I fix it? Thanks!

**Reply by yaswant** [reply]
In this situation, you can safely delete the offending entry (line 139) in your `~/.ssh/known_hosts` file.

**Comment by stevemullerworth** [comment]
It's a good idea to post a link to the output or post the full error message. The fuller error message indicates you are trying to overwrite the file. The `wb` suggests you are trying to open the file to write to it:
```

File "/projects/um1/mule/mule-2020.01.1/python-3.6.10_numpy-1.17.4/openmp/lib/mule/__init__.py", line 1411, in to_file
    with open(output_file_or_path, 'wb') as output_file:
PermissionError: [Errno 13] Permission denied: '/home/d00/hadac/cylc-run/u-cz993/share/data/ozone_redistribution/mmro3_monthly_CMIP6_2082_N96_cz993-ancil_2anc'
[FAIL] python_env python ${CYLC_SUITE_DEF_PATH}/src/contrib/redistribute_ozone.py -t $TROPOPAUSE_INPUT -r $OROGRAPHY_INPUT -d $DENSITY_INPUT -z $OZONE_INPUT -o $OZONE_OUTPUT -y $YEAR # return-code=1
2023-12-18T13:11:08Z CRITICAL - failed/EXIT
```

**Reply by mo-abodas** [reply]
Thanks. I don't understand why it's doing that, perhaps I need to change something in that app for a continuation run.
